### PR TITLE
Replace isort and flake8 with Ruff checker

### PR DESCRIPTION
**Pull request by mwtoews**
This PR replaces isort and flake8 lint tools with [Ruff](https://docs.astral.sh/ruff/) check tools. The rational is that Ruff is very fast, which has earned it's recent popularity. It also has Drop-in parity with flake8, isort, and Black.

With the transition, the old configurations are mapped to new configurations. Some of these were redundant or some ignores have no effect, so are not transitioned.

This PR does not replace Black with Ruff format, as this would incur more changed files (I'm seeking for minimal changes to source files). But a follow-up PR can replace Black with Ruff format, with the updated source files.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -60,17 +60,32 @@ extend-exclude = """
...
+
+[tool.ruff.lint]
+select = [
+    "E",  # pycodestyle error
+    "F",  # Pyflakes
+    "I",  # isort
+]
+ignore = [
+    "E722",  # do not use bare 'except'
+    "E721",  # use `is` and `is not` for type comparisons
```
Comment (fchollet) [inline pyproject.toml:74]: Do we really need to ignore this one?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! We can definitely consider this change. Please take a look at the code format check failure.

File: pyproject.toml (hunk 1 above)
Comment (mwtoews) [reply pyproject.toml:74]: Sure, I'll add another commit in this PR to resolve it.

**Comment by mwtoews** [comment]
> The rational is that Ruff is very fast

Looking at the speedup via `time bash shell/lint.sh` (and commenting out `black ...`) the results in `master` are:
```bash
$ time bash shell/lint.sh
...
real	0m3.153s
user	0m29.378s
sys	0m0.430s
```
and with this PR are significantly faster:
```bash
$ time bash shell/lint.sh
...
real	0m0.030s
user	0m0.029s
sys	0m0.053s
```

**Review by fchollet — APPROVED** [review]
LGTM, thank you

**Pull request merged**